### PR TITLE
Add D2BuildingPlacementRenderable and D2PlaceBuildingPreview

### LINKED ISSUE #110: Buildings placement
Currently buildings drawn during placement. In original only rectangle drawn.

### PR DESCRIPTION
**Pull request by evgeniysergeev**
Closes #110 

Do not draw all building sprites, just bounds rect for buildings placements

### CODE REVIEW

**Comment by TheRaffy** [comment]
title is missing a d.

**Comment by evgeniysergeev** [comment]
> title is missing a d.

not following. please provide a link to exact line in pr?

File: OpenRA.Mods.D2/Graphics/D2BuildingPlacementRenderable.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The d2 mod Developers (see AUTHORS)
```
Comment (abcdefg30) [inline OpenRA.Mods.D2/Graphics/D2BuildingPlacementRenderable.cs]: 2018?

File: OpenRA.Mods.D2/Traits/Buildings/D2PlaceBuildingPreview.cs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+		protected float t = 0.0f;
+	
+		public D2PlaceBuildingPreviewPreview(WorldRenderer wr, ActorInfo ai, D2PlaceBuildingPreviewInfo info, TypeDictionary init)
+			: base(wr, ai, info, init)
+		{
+			this.info = info;
+			this.ai = ai;
+
+			bi = ai.TraitInfoOrDefault<D2BuildingInfo>();
```
Comment (abcdefg30) [inline OpenRA.Mods.D2/Traits/Buildings/D2PlaceBuildingPreview.cs]: You are not checking if `bi` is null, so it is better to just use `.TraitInfo<D2BuildingInfo>` here.

Comment (evgeniysergeev) [reply OpenRA.Mods.D2/Traits/Buildings/D2PlaceBuildingPreview.cs]: Also added `Requires<D2BuildingInfo>`

**Comment by MustaphaTR** [comment]
resim

I feel like this should show placeable because it is placeable.

**Comment by evgeniysergeev** [comment]
> resim
> 
> I feel like this should show placeable because it is placeable.

if it is not placeable because of shroud, then it is expected. 
for other situations, when some concrete in 2x2 concrete construction is not buildable, but others is good, fixup is pushed and now it showing all 2x2 concrete placeable

**Comment by evgeniysergeev** [comment]
> resim
> 
> I feel like this should show placeable because it is placeable.

Should be fixed. re-review pr please

**Pull request merged**